### PR TITLE
Feature/endpoints

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?

https://github.com/smashingboxes/apprenticeship/tree/master/Curriculum/Backend/rails_todo
Because API endpoints are important to make a nice app useful
### What Changed?

API endpoints were defined
Seeds were added
Tests were added

### CODE REVIEW

**Comment by BrandonMathis** [comment]
You're missing some [Controller tests](https://www.relishapp.com/rspec/rspec-rails/docs/controller-specs) for this PR. When developing an API, all your endpoints must be tested to assure that they return expected JSON.

**Comment by TylerRockwell** [comment]
@BrandonMathis Thanks. I wasn't exactly sure how to get them implemented for APIs. I'll read the link and see what I can do.

**Comment by TylerRockwell** [comment]
@BrandonMathis Controller tests have been added.

File: app/models/todo.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class Todo < ActiveRecord::Base
+  def as_json
+    { id: id, title: title, is_completed: is_completed }
+  end
```
Comment (jahammo2) [inline app/models/todo.rb:4]: This is fine for now. But I'm sure Annie will be showing you [Active Model Serializers](https://github.com/rails-api/active_model_serializers) at some point.

Don't change anything yet! But rest assured that there is an even cooler way to do this

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,6 @@
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
+150.times do
+  Todo.create(title: Faker::Company.bs, is_completed: [true, false].sample)
```
Comment (jahammo2) [inline db/seeds.rb:9]: :+1:

File: spec/controllers/todos_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      get :index
+      expect(response_json).to include(
+        todos: Todo.all.map do |todo|
+          {
+            id: todo.id,
+            title: todo.title,
+            is_completed: todo.is_completed
+          }
+        end
+      )
```
Comment (jahammo2) [inline spec/controllers/todos_controller_spec.rb:20]: :+1:

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,92 +1,9 @@
...
-  # the seed, which is printed after each run.
-  #     --seed 1234
-  config.order = :random
-
-  # Seed global randomization in this process using the `--seed` CLI option.
-  # Setting this allows you to use `--seed` to deterministically reproduce
-  # test failures related to randomization by passing the same `--seed` value
-  # as the one that triggered the failure.
-  Kernel.srand config.seed
-=end
```
Comment (jahammo2) [inline spec/spec_helper.rb:91]: Down with the docs!

File: spec/support/json_support.rb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+def response_json
+  HashWithIndifferentAccess.new(JSON.parse(response.body))
```
Comment (jahammo2) [inline spec/support/json_support.rb:2]: nice

**Comment by jahammo2** [comment]
I don't see any blockers now that you have controller tests. :shipit:

File: app/models/todo.rb (hunk 1 above)
Comment (TylerRockwell) [inline app/models/todo.rb:4]: I came across AMS while I was trying to set this up, but I wasn't sure if I was supposed to use them. I'll wait for Annie to talk about them first, though. Things make so much more sense when coming from her.

File: spec/support/json_support.rb (hunk 5 above)
Comment (TylerRockwell) [inline spec/support/json_support.rb:2]: That was all thanks to Annie.

**Comment by TylerRockwell** [comment]
Thanks for the code review, @jahammo2 !

File: spec/controllers/todos_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  describe "DELETE #destroy" do
+    let(:params){ { id: Todo.last.id } }
+    let(:send_request){ delete :destroy, params }
+
+    it "should delete a specified todo and return an empty hash" do
+      expect { send_request }.to change(Todo, :count).by(-1)
+      expect(response_json).to include({})
+    end
+  end
+end
```
Comment (BrandonMathis) [inline spec/controllers/todos_controller_spec.rb:74]: These controller test look great man!

Nice work :100:

**Pull request merged**